Fight.cpp

```diff
@@ -0,0 +1,74 @@
+#include "Fight.h"
+
+#include "Item.h"
+#include "Modifier.h"
+#include "Munchkin.h"
+#include "Monster.h"
+#include "Runaway.h"
+#include "Victory.h"
+
+void Fight::start()
+{
+	//Setup initial state
+	m_result = FightResult::InProgress;
+
+	calculateMunchkinPower();
+	calculateMonsterPower();
+}
+
+void Fight::applyModifier(int choice)
+{
+	Modifier* modifier = m_munchkin->popModifier(choice);
+	if (modifier == nullptr)
+	{
+		//Give some assert/warning in debug to a user that modifier is null so something is wrong
+		return;
+	}
+
+	modifier->apply(m_munchkin, m_monster);
+
+	m_MunchkinPower += modifier->getMunchkinModifier();
+	m_monsterPower += modifier->getMonsterModifier();
+}
+
+void Fight::runawayFlow()
+{
+	Runaway* policy = m_monster->getRunawayPolicy();
+	policy->apply(m_munchkin);
+
+	m_result = FightResult::MonsterWon;
+}
+
+void Fight::victoryFlow()
+{
+	//#TODO: Implement at LEAST ONE victory policy similar to runaway policy
+	//Possible policies are:
+	//  Add new cards to hand
+	//  Generate new Outfit items???
+	//  Increase Level by 0,1,2
+
+	Victory* policy = m_monster->getVictoryPolicy();
+
+	if (policy != nullptr) {
+		policy->apply(m_munchkin);
+	}
+	else {
+		m_munchkin->updateLevelBy(1);
+	}
```

Comment:
To remove this if/else statement you can implement `default` victory policy, which will be assigned to every monster if no other was passed in c-tor. And that VictoryPolicy_Default will increase level by 1.